### ISSUE TITLE
Consider using docker images tags :latest :stable for docker images

### ISSUE THREAD

**Issue by fgrzadkowski** [kubernetes/kubernetes#20715]
During release process we use only numbered docker image versions, e.g. `v1.1.7` or `v.1.2.0-alpha.7`. From the user perspective it would be easier to use `gcr.io/google_containers/hyperkube-amd64:stable` for `v1.1.7` or `gcr.io/google_containers/hyperkube-amd64:latest` for `v1.2.0-alpha.7`. This will allow us to avoid hardcoding versions in various documents like docker getting started guide.

Forked from #16087

@brendandburns @luxas @dalanlan @ihmccreery @david-mcmahon

**Comment by luxas** [comment]
SGTM

But there should be both the `stable` and `v1.x.y` tag, so when `v1.x.z` is released, that one becomes `stable` but `v1.x.y` is still usable.

Could `latest` be replaced with something else? `latest` is docker's predefined value and equals to just `gcr.io/google_containers/hyperkube-amd64`, so would e.g. `canary` or `alpha` sound better?

**Comment by fgrzadkowski** [comment]
I agree we should use `v1` and `stable` tags. Using `alpha` or `beta` sounds good to me.

**Comment by luxas** [comment]
OK, so these tags should be used then:
 - The real version: e.g. `v1.3.0-alpha.3`, `v1.2.3`, `v2.0.1`, `v2.1.0-alpha.7`, `v2.1.0-beta.2`
 - A `v1` (== `stable` on `v1.x`) tag? => `v1.2.3`
 - A `v2` (== `stable` on `v2.x`) tag? => `v2.0.1`
 - A `stable` tag: `v2.0.1`
 - An `alpha` tag => `v2.1.0-alpha.7`
 - A `beta` tag => `v2.1.0-beta.2`

Is that what you meant? The question is what should happen when `v2.1.0` is released... I understand it as at the same time a `v2.2.0-alpha.0` will be cut, so that one becomes `alpha`.

I'm not really sure if we should have the `beta` tag.
How about `unstable` instead of `alpha`? Just a suggestion though...

**Comment by fgrzadkowski** [comment]
We do have different release process for `alpha` and `beta`, so I'd keep it that way. Also it'd more consistent IMO.

**Comment by luxas** [comment]
OK. One example: `v1.2.0-beta.0` that's released soon will be `beta`, but what will happen when `v1.2.0` is released? The next `beta` version will then be `v1.3.0-beta.0`, which can take ~3 months. I mean that's a very long time.

The other option is something like `canary`, `unstable`, which always points to the latest release, and may be `v1.2.0-alpha.8` or `v1.2.0-beta.1`. You just know that this is always the latest

It's a bit hard to describe this things, and it's a bit complex too. WDYT about the `v1` and `v2` tags, are they necessary?

**Comment by fgrzadkowski** [comment]
That's a good point. In that case we might consider not using `beta` at all. I'd still use `alpha` and not introduce new names like `canary`.

**Comment by luxas** [comment]
`alpha` sgtm
Should we ditch the `v1` and `v2` too? I think many, many other things have changed until that, so this discussion won't be relevant then.

TL;DR;
`stable` tag for versions like `v1.1.2`, `v1.2.3` and `v1.3.0`
`alpha` tag for versions like `v1.2.0-alpha.8`, `v1.3.0-alpha.0` (which in fact **should** be is equal to `v1.2.0-beta.0`, but this hasn't been always the case), and  `v1.3.0-alpha.7`

I think this will be an improvement
